org.eclipse.scout.rt.server.commons/src/main/java/org/eclipse/scout/rt/server/commons/authentication/TrivialAccessController.java

```diff
@@ -254,4 +257,83 @@ public TrivialAuthConfig withLoginPageInstalled(final boolean loginPageInstalled
       return this;
     }
   }
+
+  /**
+   * Checks if the request is authenticated by either a valid subject or a valid principal.
+   *
+   * @param request
+   *     Http request
+   * @return {@code true} if request is authenticated. Otherwise {@code false}.
+   */
+  protected boolean isAuthenticatedRequest(HttpServletRequest request) {
+    return ObjectUtility.isOneOf(getAuthenticationStatus(request).getStatus(), AuthenticationStatusType.SUBJECT_VALID, AuthenticationStatusType.PRINCIPAL_VALID);
```

Comment:
This method should get an AuthenticationStatusType instead of the request and only do the isOneOf. Otherwise getAuthenticationStatus(request) must be called twice.